tests/lxd/test_lxd_backend.cpp

```diff
@@ -0,0 +1,602 @@
+/*
+ * Copyright (C) 2020 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <src/platform/backends/lxd/lxd_virtual_machine_factory.h>
+
+#include "mock_lxd_server_responses.h"
+#include "tests/local_socket_server_test_fixture.h"
+#include "tests/mock_logger.h"
+#include "tests/mock_status_monitor.h"
+#include "tests/stub_status_monitor.h"
+#include "tests/temp_dir.h"
+
+#include <multipass/logging/log.h>
+#include <multipass/memory_size.h>
+#include <multipass/platform.h>
+#include <multipass/version.h>
+#include <multipass/virtual_machine_description.h>
+
+#include <QString>
+#include <QUrl>
+
+#include <gmock/gmock.h>
+
+namespace mp = multipass;
+namespace mpl = multipass::logging;
+namespace mpt = multipass::test;
+
+using namespace testing;
+
+namespace
+{
+using LXDInstanceStatusParamType = std::pair<QByteArray, mp::VirtualMachine::State>;
+
+struct LXDBackend : public Test
+{
+    LXDBackend()
+        : socket_path{QString("%1/test_socket").arg(data_dir.path())},
+          test_server{mpt::MockLocalSocketServer(socket_path)},
```

Comment:
Should we not test this on the access manager level?

Reply:
We already unit test the access manager itself in other tests, so I think mocking it as high up as possible is the best way to go for testing the backend.  I sense you disagree with this approach, so am I missing some benefits if we use the access manger as well?

Reply:
I mean to _mock_ the access manager, rather than talking through a test socket (which was the correct thing to do for testing the manager itself).

Reply:
Duh on my end.  I will do that to simplify this.

Reply:
Looking at this a bit more, what will need to happen is to make `LocalAccessManager` a non-`final` class and then create a mock `LocalAccessManager` class and have it respond how we want.  Then we'll need to modify `LXDVirtualMachine` and `LXDVMImageVault` to accept a `LocalAccessManager` object (ie, the mocked object) and not create one in the ctor.

Then we can get rid of the `MockLocalSocketServer` and just put that logic back into the `test_local_network_access_manager.cpp` file.

Reply:
Even some more diving into this and really, doing it the current way where we talk over a local socket to a "mocked" server is the most straightforward way to go.  If we went the route of mocking `NetworkAccessManager` itself, we'll basically need to reimplement that along with `LocalSocketReply` and that doesn't really make much sense.  I think leaving it the way it is is the right thing to do in order not to introduce much more complexity in testing for the sake of not using a temporary local socket.